src/components/grid-view/GridView.js

```diff
@@ -38,36 +39,47 @@ class GridView extends React.Component<Props> {
     });
 
     componentDidUpdate(prevProps: Props) {
-        const { columnCount, width } = this.props;
+        const { columnCount, currentCollection, width } = this.props;
 
         // The React Virtualized Table must be notified when either the cached
         // row sizes or the parent width change. If omitted, rows are sized
         // incorrectly resulting in gaps or content overlap.
-        if (columnCount !== prevProps.columnCount || width !== prevProps.width) {
+        if (
+            columnCount !== prevProps.columnCount ||
+            currentCollection.id !== prevProps.currentCollection.id ||
+            width !== prevProps.width
+        ) {
             this.cache.clearAll();
             this.forceUpdate();
         }
     }
 
     cellRenderer = ({ dataKey, parent, rowIndex }: TableCellRendererParams) => {
-        const { columnCount, currentCollection, slotRenderer } = this.props;
+        const { columnCount, currentCollection, onItemClick, slotRenderer } = this.props;
         const count = getProp(currentCollection, 'items.length', 0);
         const contents = [];
 
         const startingIndex = rowIndex * columnCount;
         const maxSlotIndex = Math.min(startingIndex + columnCount, count);
 
         for (let slotIndex = startingIndex; slotIndex < maxSlotIndex; slotIndex += 1) {
+            const item = getProp(currentCollection, `items[${slotIndex}]`);
+            const { id, selected } = item;
+
             // using item's id as key is important for renrendering.  React Virtualized Table rerenders
             // on every 1px scroll, so using improper key would lead to image flickering in each
             // card of the grid view when scrolling.
             contents.push(
-                <div
-                    key={getProp(currentCollection, `items[${slotIndex}].id`) || uniqueId('bdl-GridView-slot')}
-                    className="bdl-GridView-slot"
+                <button
```

Comment:
If I remember correctly, there can be a dropdown menu toggle button inside of each cell. This will lead to buttons inside of buttons, which isn't valid markup. It seems like the cell itself should remain a div and the clickable elements inside it should be buttons.